impl/src/main/java/org/ehcache/impl/internal/executor/PooledExecutionService.java

```diff
@@ -129,21 +112,27 @@ public void start(ServiceProvider<Service> serviceProvider) {
     running = true;
   }
 
+  /**
+   * Stop the service. Underlying executors will be stopped calling {@code shutdownNow}. Pending tasks are discarded
+   */
   @Override
   public void stop() {
     LOGGER.debug("Shutting down PooledExecutionService");
     running = false;
-    //scheduledExecutor.shutdown();
+    scheduledExecutor.shutdownNow();
     for (Iterator<Entry<String, ThreadPoolExecutor>> it = pools.entrySet().iterator(); it.hasNext(); ) {
       Entry<String, ThreadPoolExecutor> e = it.next();
       try {
-        if (e.getKey() != null) {
-          destroyPool(e.getKey(), e.getValue());
-        }
+        destroyPool(e.getKey(), e.getValue());
       } finally {
         it.remove();
       }
     }
+    try {
+      scheduledExecutor.awaitTermination(30, SECONDS);
```

Comment:
Shouldn't this be in a similar loop to the one used in the `destroyPool` method?